File: output/xCodeEval_8000/C++/Go/18f4c797aaf57b3947a6000d6b07af4c.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

const (
	N = 1000009
	M = 10009
	Z = 26
)

var (
	A    [N]int
	buf  [Z]byte
	Q    [M]int
	have = make(map[int]bool)
	ans  = make(map[int]int)
	n, m int
)

func push(x int) {
	if !have[x] {
		have[x] = true
	} else {
		j := 0
		for ; stk[j] != x; j++ {
		}
		for ; j < nn-1; j++ {
			stk[j] = stk[j+1]
		}
		stk[nn-1] = x
	}
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	n = 0
	for {
		buf[0], _ = reader.ReadByte()
		if buf[0] == '\n' {
			break
		}
		A[n] = int(buf[0] - 'a')
		n++
	}
	A[n] = Z

	m = 0
	for {
		buf[0], _ = reader.ReadByte()
		if buf[0] == '\n' {
			break
		}
		if buf[0] == ' ' {
			continue
		}
		Q[m] = 0
		for {
			Q[m] |= 1 << (buf[0] - 'a')
			buf[0], _ = reader.ReadByte()
			if buf[0] == '\n' {
				break
			}
			if buf[0] == ' ' {
				break
			}
		}
		have[Q[m]] = true
		m++
	}

	nn = 0
	for i := 0; i < n; i++ {
		push(A[i])
		s := 0
		for j := nn; j >= 0; j-- {
			s |= 1 << stk[j]
			if s&(1<<A[i+1]) != 0 {
				break
			}
			if have[s] {
				ans[s]++
			}
		}
	}

	for i := 0; i < m; i++ {
		fmt.Println(ans[Q[i]])
	}
}

File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/C/18f4c797aaf57b3947a6000d6b07af4c.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <set>

#define REP(i, n) for(int i = 0; i < (n); ++i)
#define REP_C(i, n) for(int i = 0; i < (n); ++i)
#define REP_S(cur, str) for(char *cur = (str); *cur; ++cur)
#define DWN(i, n, m) for(int i = (n) - 1; i >= (m); --i)

typedef long long LL;
typedef std::set<int> SI;
typedef std::map<int, int> MII;

const int N = 1000009, M = 10009, Z = 26;

char A[N], buf[Z]; 
int Q[M]; 
SI have; 
MII ans;
int n, m;

int stk[Z], nn; 
bool vis[Z]; 

void push(int x) {
    if (!vis[x]) {
        vis[stk[nn++] = x] = true;
    } else {
        int j; 
        for(j = 0;; j++) if(stk[j] == x) break;
        for(; j < nn - 1; j++) stk[j] = stk[j + 1]; 
        stk[nn - 1] = x;
    }
}

int main() {
    freopen("in.txt", "r", stdin);

    REP_C(i, n = strlen(fgets(A, N, stdin))) A[i] -= 'a'; 
    A[n] = Z;

    REP_C(i, m) {
        fgets(buf, Z, stdin); 
        REP_S(cur, buf) Q[i] |= (1 << (*cur - 'a'));
        have.insert(Q[i]);
    }

    REP(i, n) {
        push(A[i]); 
        int s = 0; 
        DWN(j, nn, 0) {
            s |= (1 << stk[j]); 
            if ((s & (1 << A[i + 1]))) break;
            if (have.count(s)) ++ans[s];
        }
    }

    REP(i, m) printf("%d\n", ans[Q[i]]);
}
